www/include/configuration/configObject/host_categories/DB-Func.php

```diff
@@ -94,86 +95,163 @@ function enableHostCategoriesInDB($hc_id = null, $hc_arr = array())
+                        break;
+                    case 'hc_alias':
+                        $value2 = filter_var($value2, FILTER_SANITIZE_STRING);
+                        $bindParams[':hc_alias'] = [\PDO::PARAM_STR => $value2];
+                        break;
+                    case 'level':
+                        $value2 = filter_var($value2, FILTER_VALIDATE_INT);
+                        if ($value2) {
+                            $bindParams[':level'] = [\PDO::PARAM_INT => $value2];
+                            $level = true;
+                        } else {
+                            $bindParams[':level'] = [\PDO::PARAM_NULL => null];
+                        }
+                        break;
+                    case 'icon_id':
+                        $value2 = filter_var($value2, FILTER_VALIDATE_INT);
+                        $value2
+                            ? $bindParams[':icon_id'] = [\PDO::PARAM_INT => $value2]
+                            : $bindParams[':icon_id'] = [\PDO::PARAM_NULL => null];
+                        break;
+                    case 'hc_comment':
+                        $value2 = filter_var($value2, FILTER_SANITIZE_STRING);
+                        $value2
+                            ? $bindParams[':hc_comment'] = [\PDO::PARAM_STR => $value2]
+                            : $bindParams[':hc_comment'] = [\PDO::PARAM_NULL => null];
+                        break;
+                    case 'hc_activate':
+                        $value2 = filter_var($value2, FILTER_VALIDATE_REGEXP, [
+                            "options" => [
+                                "regexp" => "/^0|1$/"
+                            ]
+                        ]);
+                        $value2
+                            ? $bindParams[':hc_activate'] = [\PDO::PARAM_STR => $value2]
+                            : $bindParams[':hc_activate'] = [\PDO::PARAM_STR => "0"];
+                        break;
+                }
+
                 if ($key2 != "hc_id") {
                     $fields[$key2] = $value2;
                 }
-                if ($key2 == "level" && $value2 != "") {
-                    $level = true;
-                }
+                $val
+                    ? $val .= ($value2 != null ? (", '" . $value2 . "'") : ", NULL")
+                    : $val .= ($value2 != null ? ("'" . $value2 . "'") : "NULL");
             }
             $fields["hc_name"] = $hc_name;
             if (testHostCategorieExistence($hc_name)) {
-                $val ? $rq = "INSERT INTO hostcategories VALUES (" . $val . ")" : $rq = null;
-                $pearDB->query($rq);
+                $val
+                    ? $query = "
+                        INSERT INTO hostcategories
+                        VALUES (NULL, :hc_name, :hc_alias, :level, :icon_id, :hc_comment, :hc_activate)"
+                    : $query = null;
+                $statement = $pearDB->prepare($query);
```

Comment:
Are you trying to prepare and bind values on a $query = null (https://github.com/centreon/centreon/pull/8914/files#diff-0bb59ae13ef1e73cccc8b91988afb651R234)?